coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/KafkaMetricHistogram.java

```diff
@@ -125,6 +127,6 @@ public List<NamedMeasurable> stats() {
 
     @Override
     public void record(MetricConfig config, double value, long timeMs) {
-        hdrHistogram.record((long) value);
+        hdrHistogram.record(Math.min(highestTrackableValue, (long) value));
```

Comment:
Should we add a unit test for this change in KafkaMetricHistogramTest
?

Reply:
out of curiosity, `HdrHistogram` is the wrap of HdrHistogram API, so maybe we can move this protection into `HdrHistogram`?